src/App.js

```diff
@@ -27,6 +28,13 @@ function App() {
         path="/stories/:userId/"
         render={(match) => <UserStories userId={match.match.params.userId} />}
       ></Route>
+      <Route
+        path={['/:userName/', '/:userName/channel/', '/:userName/tagged/']}
+        exact
+        render={(match) => (
+          <UserProfile userName={match.match.params.userName} />
+        )}
+      ></Route>
```

Comment:
Since this route accepts `/anything` (see the first path) you have to put it last in the Switch. Let me know if you don't understand this.